DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                                LearningHubAuthId = @learningHubAuthId,
+                                HasDismissedLhLoginWarning = @hasDismissedLhLoginWarning,
+                                EmailVerified = GETUTCDATE(),
+                                DetailsLastChecked = CASE WHEN @detailsMatched = 0 OR DetailsLastChecked IS NULL THEN NULL ELSE GETUTCDATE()
+                            WHERE ID = @userId",
+                new
+                {
+                    userId = existingUserWithEmail.Id,
+                    passwordHash = string.IsNullOrEmpty(existingUserWithEmail.PasswordHash)
+                        ? delegateAccount.OldPassword ?? ""
+                        : existingUserWithEmail.PasswordHash,
+                    firstName = string.IsNullOrEmpty(existingUserWithEmail.FirstName)
+                        ? delegateAccount.FirstName_deprecated ?? ""
+                        : existingUserWithEmail.FirstName,
+                    lastName = string.IsNullOrEmpty(existingUserWithEmail.LastName)
+                        ? delegateAccount.LastName_deprecated
+                        : existingUserWithEmail.LastName,
+                    jobGroupId = existingUserWithEmail.JobGroupId == 10
+                        ? delegateAccount.JobGroupId_deprecated
+                        : existingUserWithEmail.JobGroupId,
+                    professionalRegistrationNumber =
+                        string.IsNullOrEmpty(existingUserWithEmail.ProfessionalRegistrationNumber)
+                            ? delegateAccount.ProfessionalRegistrationNumber_deprecated
+                            : existingUserWithEmail.ProfessionalRegistrationNumber,
+                    profileImage = existingUserWithEmail.ProfileImage ?? delegateAccount.ProfileImage_deprecated,
+                    active = existingUserWithEmail.Active || delegateAccount.Active,
+                    resetPasswordId = existingUserWithEmail.ResetPasswordId ??
+                                      delegateAccount.ResetPasswordId_deprecated,
+                    hasBeenPromptedForPrn = existingUserWithEmail.HasBeenPromptedForPrn ||
+                                            delegateAccount.HasBeenPromptedForPrn_deprecated,
+                    learningHubAuthId = existingUserWithEmail.LearningHubAuthId ??
+                                        delegateAccount.LearningHubAuthId_deprecated,
+                    hasDismissedLhLoginWarning = existingUserWithEmail.HasDismissedLhLoginWarning ||
+                                                 delegateAccount.HasDismissedLhLoginWarning_deprecated,
+                    detailsMatch = DoesDelegateAccountMatchExistingUser(delegateAccount, existingUserWithEmail, allJobGroupsMatch)
+                }
+            );
+
+            UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(
+                connection,
+                existingUserWithEmail.Id,
+                delegateAccount.Id
+            );

```

Comment:
I think we can do this right at the end of the migration rather than after every individual update.
Because we're running this method on all inserts, the only remaining delegates with null userIds at the end of the migration will have an email address matching their user record.
So then you can do a single query for all of them as you've done for admins. Should save us a lot of time.